### PR TITLE
Add paste mode support

### PR DESCRIPTION
**Pull request by gnuyent**
This PR adds paste mode support. Every mode should be supported with the proper highlighting. Let me know if it this is good!

WindowsTerminal_6wkZUH5ZBB
WindowsTerminal_dGcb3l4hA7
WindowsTerminal_SfHVKFTNjW
WindowsTerminal_aAKCANVd6O

### CODE REVIEW

**Comment by shadmansaleh** [comment]
I'm not sure what paste mode does but neovim help pages say paste option is obsolete
from help pages:
>'paste'                 boolean (default off)
                        global
        This option is obsolete; bracketed-paste-mode is built-in.

Can you say what you're using paste option for ?

**Comment by gnuyent** [comment]
Yes, I use Windows Terminal which [currently doesn't support bracketed paste](https://github.com/microsoft/Terminal/issues/395). As a workaround, I use paste mode in neovim.

**Comment by hoob3rt** [comment]
I don't think we should support obsolete features since hardly anyone will even notice this change and it only adds unnecessary code to the codebase :)

**Comment by skyline75489** [comment]
Hey just FYI, the next version Windows Terminal Preview will have initial support for bracketed paste.

**Comment by gnuyent** [comment]
> Hey just FYI, the next version Windows Terminal Preview will have initial support for bracketed paste. 

Awesome, I look forward to avoiding paste mode!

**Comment by Grazfather** [comment]
Thanks for this PR anyway, found this in search since I wanted to add paste myself, and discovered I [mostly] don't need it anymore.

tl;dr for others who search it: If your terminal is smart, when you paste it'll surround it with `\e[200~` and  `\e[201~`, which tells the app, e.g. Neovim not to auto format.

I still have the issue, however, when I paste using tmux, since it doesn't know to do this. Note that paste mode DOES still work in Neovim, despite it being obsoleted.
